File: src/linkedlist/lc_1669_merge_between_linked_lists/merge_between_linked_lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeInBetween(struct ListNode* list1, int a, int b, struct ListNode* list2){
    struct ListNode* temp;
    struct ListNode* temp1 = list1;
    struct ListNode* temp2 = list2;
    int count = 0;
    while(temp1->next != NULL)
    {
        if(count == a - 1)
        {
            temp = temp1;
            while(count < b)
            {
                temp = temp->next;
                count++;
            }
            temp1->next = list2;
            while(temp2->next != NULL)
            {
                temp2 = temp2-> next;
            }
            temp2->next = temp->next;
            free(temp);
            
        }
        else{
            count++;
            temp1 = temp1->next;
        }
        
    }
    return list1;
}

File: src/linkedlist/lc_1669_merge_between_linked_lists/merge_between_linked_lists.py
from listNode import ListNode
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        dummy = ListNode(-1, list1)
        count = 0
        temp2 = list2
        #temp2 at end of list 2
        while temp2.next:
            temp2 = temp2.next
        tempa = None
        tempb = None
        temp = list1
        while temp:
            if count == a - 1:
                tempa = temp
                break
            temp = temp.next
            count += 1
        while temp:
            if count == b:
                tempb = temp
                break
            temp = temp.next
            count += 1
        tempa.next = list2
        temp2.next = tempb.next
        return dummy.next
            
        